src/Stan/Partial.hs

```diff
@@ -31,26 +31,33 @@ findPartials = mapMaybe toWarning . lNames
     toWarning (L l name) =
         let modName = moduleName $ nameModule name
             occName = nameOccName name
-        in
-        if checkName modName occName
-            then Just $ Partial (occNameString occName) l
-            else Nothing
+        in guard (ifPartial modName occName && isRealSrcSpan l)
+               *> Just (Partial (occNameString occName) $ getReal l)
```

Comment:
Programming pattern like `if isJust foo then fromJust foo` is considered bad. Here you can use `do`-notation desugaring and the fact that pattern-match failure results in call to `fail` function. In case of `Maybe` function `fail` just returns `Nothing`. So you can safely write code like this:

```haskell
foo = do
    RealSrcSpan src <- Just l
    guard $ isPartial modName occName
    Just {- or pure -} $ Partial (occNameString occName) src
``` 

Alternatively you can change type of `getReal` function to be `getReal :: SrcSpan -> Maybe RealSrcSpan` but approach with `fail` allows you to not write extra function.